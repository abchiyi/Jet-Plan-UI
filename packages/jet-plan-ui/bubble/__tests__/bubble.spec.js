import { mount, shallowMount } from '@vue/test-utils'
import babble from '../bubble.vue'

describe('Babble', async () => {
    it('Node map', async () => {
        const wrapper = mount(babble)
        expect(wrapper.html()).toMatchSnapshot()

        await wrapper.setProps({ message: 'hello' })
        expect(wrapper.html()).toMatchSnapshot()

    })

    it('Name in map', () => {
        const wrapper = mount(babble)
        expect(wrapper.classes()).toContain(babble.name)
    })

    it('Message', async () => {
        const wrapper = mount(babble)
        await wrapper.setProps(
            {
                message: 'hello'
            }
        )

        expect(wrapper.find('j-button').html()).toContain('hello')


    }
    )


})
