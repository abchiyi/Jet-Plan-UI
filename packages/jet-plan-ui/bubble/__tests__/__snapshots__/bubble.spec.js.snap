// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Babble Node map 1`] = `
<div class="j-row" style="display: inline-block;">
  <div class="j-base-action j-bubble" ontouchstart="function () { [native code] }" ontouchcancel="function () { [native code] }" ontouchend="function () { [native code] }">
    <div class="j-row">
      <transition-stub>
        <div class="j-base-action j-action-feedback shape j-button button bubble top" ontouchstart="function () { [native code] }" ontouchcancel="function () { [native code] }" ontouchend="function () { [native code] }" style="--color-focus: var(--mask); --color-hover: var(--mask); --mask-opacity: 0.5; display: none; box-shadow: 
    0 4px
    6px
    0.4px
    var(--shadow);">hello<transition-group-stub></transition-group-stub>
        </div>
      </transition-stub>
    </div>
  </div>
</div>
`;

exports[`Babble Node map 2`] = `
<div class="j-row" style="display: inline-block;">
  <div class="j-base-action j-bubble" ontouchstart="function () { [native code] }" ontouchcancel="function () { [native code] }" ontouchend="function () { [native code] }">
    <div class="j-row">
      <transition-stub>
        <div class="j-base-action j-action-feedback shape j-button button bubble top" ontouchstart="function () { [native code] }" ontouchcancel="function () { [native code] }" ontouchend="function () { [native code] }" style="--color-focus: var(--mask); --color-hover: var(--mask); --mask-opacity: 0.5; display: none; box-shadow: 
    0 4px
    6px
    0.4px
    var(--shadow); --base-text-color: var(--text-dark);">hello<transition-group-stub></transition-group-stub>
        </div>
      </transition-stub>
    </div>
  </div>
</div>
`;

exports[`Babble Node map 3`] = `
<div class="j-row" style="display: inline-block;">
  <div class="j-base-action j-bubble" ontouchstart="function () { [native code] }" ontouchcancel="function () { [native code] }" ontouchend="function () { [native code] }">
    <div class="j-row">
      <transition-stub>
        <div class="j-base-action j-action-feedback shape j-button button bubble top" ontouchstart="function () { [native code] }" ontouchcancel="function () { [native code] }" ontouchend="function () { [native code] }" style="--color-focus: var(--mask); --color-hover: var(--mask); --mask-opacity: 0.5; display: none; box-shadow: 
    0 4px
    6px
    0.4px
    var(--shadow); --base-text-color: var(--text-dark);">hello<transition-group-stub></transition-group-stub>
        </div>
      </transition-stub>
    </div>
  </div>
</div>
`;

exports[`Babble Node map 4`] = `
<div class="j-row" style="display: inline-block;">
  <div class="j-base-action j-bubble" ontouchstart="function () { [native code] }" ontouchcancel="function () { [native code] }" ontouchend="function () { [native code] }">
    <div class="j-row">
      <transition-stub>
        <div class="j-base-action j-action-feedback shape j-button button bubble top" ontouchstart="function () { [native code] }" ontouchcancel="function () { [native code] }" ontouchend="function () { [native code] }" style="--color-focus: var(--mask); --color-hover: var(--mask); --mask-opacity: 0.5; display: none; box-shadow: 
    0 4px
    6px
    0.4px
    var(--shadow); --base-text-color: var(--text-dark);">hello<transition-group-stub></transition-group-stub>
        </div>
      </transition-stub>
    </div>
  </div>
</div>
`;

exports[`Babble Node map 5`] = `
<div class="j-row" style="display: inline-block;">
  <div class="j-base-action j-bubble" ontouchstart="function () { [native code] }" ontouchcancel="function () { [native code] }" ontouchend="function () { [native code] }">
    <div class="j-row">
      <transition-stub>
        <div class="j-base-action j-action-feedback shape j-button button bubble top" ontouchstart="function () { [native code] }" ontouchcancel="function () { [native code] }" ontouchend="function () { [native code] }" style="--color-focus: var(--mask); --color-hover: var(--mask); --mask-opacity: 0.5; display: none; box-shadow: 
    0 4px
    6px
    0.4px
    var(--shadow); --base-text-color: var(--text-dark);">hello<transition-group-stub></transition-group-stub>
        </div>
      </transition-stub>
    </div>
  </div>
</div>
`;

exports[`Babble Node map 6`] = `
<div class="j-row" style="display: inline-block;">
  <div class="j-base-action j-bubble" ontouchstart="function () { [native code] }" ontouchcancel="function () { [native code] }" ontouchend="function () { [native code] }">
    <div class="j-row">
      <transition-stub>
        <div class="j-base-action j-action-feedback shape j-button button bubble top" ontouchstart="function () { [native code] }" ontouchcancel="function () { [native code] }" ontouchend="function () { [native code] }" style="--color-focus: var(--mask); --color-hover: var(--mask); --mask-opacity: 0.5; display: none; box-shadow: 
    0 4px
    6px
    0.4px
    var(--shadow); --base-text-color: var(--text-dark);">hello<transition-group-stub></transition-group-stub>
        </div>
      </transition-stub>
    </div>
  </div>
</div>
`;

exports[`Babble Node map 7`] = `
<div class="j-row" style="display: inline-block;">
  <div class="j-base-action j-bubble" ontouchstart="function () { [native code] }" ontouchcancel="function () { [native code] }" ontouchend="function () { [native code] }">
    <div class="j-row">
      <transition-stub>
        <div class="j-base-action j-action-feedback shape j-button button bubble top" ontouchstart="function () { [native code] }" ontouchcancel="function () { [native code] }" ontouchend="function () { [native code] }" style="--color-focus: var(--mask); --color-hover: var(--mask); --mask-opacity: 0.5; display: none; box-shadow: 
    0 4px
    6px
    0.4px
    var(--shadow); --base-text-color: var(--text-dark);">hello<transition-group-stub></transition-group-stub>
        </div>
      </transition-stub>
    </div>
  </div>
</div>
`;

exports[`Babble Node map 8`] = `
<div class="j-row" style="display: inline-block;">
  <div class="j-base-action j-bubble" ontouchstart="function () { [native code] }" ontouchcancel="function () { [native code] }" ontouchend="function () { [native code] }">
    <div class="j-row">
      <transition-stub>
        <div class="j-base-action j-action-feedback shape j-button button bubble top" ontouchstart="function () { [native code] }" ontouchcancel="function () { [native code] }" ontouchend="function () { [native code] }" style="--color-focus: var(--mask); --color-hover: var(--mask); --mask-opacity: 0.5; display: none; box-shadow: 
    0 4px
    6px
    0.4px
    var(--shadow); --base-text-color: var(--text-dark);">hello<transition-group-stub></transition-group-stub>
        </div>
      </transition-stub>
    </div>
  </div>
</div>
`;

exports[`Babble Node map 9`] = `
<div class="j-row" style="display: inline-block;">
  <div class="j-base-action j-bubble" ontouchstart="function () { [native code] }" ontouchcancel="function () { [native code] }" ontouchend="function () { [native code] }">
    <div class="j-row">
      <transition-stub>
        <div class="j-base-action j-action-feedback shape j-button button bubble top" ontouchstart="function () { [native code] }" ontouchcancel="function () { [native code] }" ontouchend="function () { [native code] }" style="--color-focus: var(--mask); --color-hover: var(--mask); --mask-opacity: 0.5; display: none; box-shadow: 
    0 4px
    6px
    0.4px
    var(--shadow); --base-text-color: var(--text-dark);">hello<transition-group-stub></transition-group-stub>
        </div>
      </transition-stub>
    </div>
  </div>
</div>
`;

exports[`Babble Node map 10`] = `
<div class="j-row" style="display: inline-block;">
  <div class="j-base-action j-bubble" ontouchstart="function () { [native code] }" ontouchcancel="function () { [native code] }" ontouchend="function () { [native code] }">
    <div class="j-row">
      <transition-stub>
        <div class="j-base-action j-action-feedback shape j-button button bubble top" ontouchstart="function () { [native code] }" ontouchcancel="function () { [native code] }" ontouchend="function () { [native code] }" style="--color-focus: var(--mask); --color-hover: var(--mask); --mask-opacity: 0.5; display: none; box-shadow: 
    0 4px
    6px
    0.4px
    var(--shadow); --base-text-color: var(--text-dark);">hello<transition-group-stub></transition-group-stub>
        </div>
      </transition-stub>
    </div>
  </div>
</div>
`;

exports[`Babble Node map 11`] = `
<div class="j-row" style="display: inline-block;">
  <div class="j-base-action j-bubble" ontouchstart="function () { [native code] }" ontouchcancel="function () { [native code] }" ontouchend="function () { [native code] }">
    <div class="j-row">
      <transition-stub>
        <div class="j-base-action j-action-feedback shape j-button button bubble top" ontouchstart="function () { [native code] }" ontouchcancel="function () { [native code] }" ontouchend="function () { [native code] }" style="--color-focus: var(--mask); --color-hover: var(--mask); --mask-opacity: 0.5; display: none; box-shadow: 
    0 4px
    6px
    0.4px
    var(--shadow); --base-text-color: var(--text-dark);">hello<transition-group-stub></transition-group-stub>
        </div>
      </transition-stub>
    </div>
  </div>
</div>
`;

exports[`Babble Node map 12`] = `
<div class="j-row" style="display: inline-block;">
  <div class="j-base-action j-bubble" ontouchstart="function () { [native code] }" ontouchcancel="function () { [native code] }" ontouchend="function () { [native code] }">
    <div class="j-row">
      <transition-stub>
        <div class="j-base-action j-action-feedback shape j-button button bubble top" ontouchstart="function () { [native code] }" ontouchcancel="function () { [native code] }" ontouchend="function () { [native code] }" style="--color-focus: var(--mask); --color-hover: var(--mask); --mask-opacity: 0.5; display: none; box-shadow: 
    0 4px
    6px
    0.4px
    var(--shadow); --base-text-color: var(--text-dark);">hello<transition-group-stub></transition-group-stub>
        </div>
      </transition-stub>
    </div>
  </div>
</div>
`;
